receipt_dynamo/receipt_dynamo/data/_compaction_run.py

```
@@ -0,0 +1,188 @@
+    def mark_compaction_run_completed(
+        self,
+        image_id: str,
+        receipt_id: int,
+        run_id: str,
+        collection: str,
+        merged_vectors: int = 0,
+    ) -> None:
+        """Mark a collection state as COMPLETED and set finished_at + merged count."""
+        run = self.get_compaction_run(image_id, receipt_id, run_id)
+        if run is None:
+            raise EntityNotFoundError("CompactionRun not found")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)